lib/tasks/db.rake

```diff
@@ -40,16 +43,52 @@ namespace :db do
     con.execute("ALTER ROLE #{aact_superuser} IN DATABASE #{aact_back_db} SET SEARCH_PATH TO ctgov, support, public, ctgov_beta;")
   end
 
+  task single_row_comparison: [:environment] do
+    # hardcoded table names with one row per study from excel file
+    file_table_names=['brief_summaries', 'calculated_values', 'categories',
+                      'designs', 'detailed_descriptions', 'eligibilities',
+                      'participant_flows', 'studies']
+
+    # finds table_names and their columns according to list in the file_table_names
+    sql = "SELECT t.table_name,
+                  c.column_name
+           FROM information_schema.tables AS t
+           INNER JOIN information_schema.columns AS c on c.table_name = t.table_name
+                                                  AND c.table_schema = t.table_schema
+           WHERE t.table_schema= 'ctgov' AND c.table_name IN (#{file_table_names.map { |e| "'#{e}'" }.join(', ')});"
+    all_columns = ActiveRecord::Base.connection.execute(sql).to_a
+
+    # comparing data in each column and adding it to a csv file if there's data mismatching
+    all_columns.each do |column|
+      query = "SELECT T.nct_id,
+                      T.#{column["column_name"]} as ctgov_#{column["column_name"]},
+                      BT.#{column["column_name"]} as ctgov_beta_#{column["column_name"]}
+              FROM ctgov.#{column["table_name"]} T
+              JOIN ctgov_beta.#{column["table_name"]} BT ON T.nct_id = BT.nct_id
+              WHERE T.#{column["column_name"]} != BT.#{column["column_name"]}
+              OR (T.#{column["column_name"]} IS NOT NULL AND BT.#{column["column_name"]} IS NULL)
+              OR (T.#{column["column_name"]} IS NULL AND BT.#{column["column_name"]} IS NOT NULL);"
+      result =ActiveRecord::Base.connection.execute(query).to_a
+
+      if result.count > 0
+        file = "./public/static/beta_comparison/#{column['table_name']}-#{column["column_name"]}.csv"
```

Comment:
put this in the same folder as the study counts folder `beta_differences/single-row`. Use the `FileManager` model method to get the folder name.